sdk/search/azure-search/src/main/java/com/azure/search/SearchClientBuilder.java

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search;
+
+import com.azure.core.annotation.ServiceClientBuilder;
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpPipelineBuilder;
+import com.azure.core.http.policy.HttpLogOptions;
+import com.azure.core.http.policy.HttpLoggingPolicy;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.http.policy.UserAgentPolicy;
+import com.azure.core.util.Configuration;
+import com.azure.core.util.CoreUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This is the base class to be extended by client builders: {@link SearchIndexClientBuilder SearchIndexClientBuilder}
+ * and {@link SearchServiceClientBuilder SearchServiceClientBuilder}, and it includes common variables and methods
+ * used by both builders
+ */
+@ServiceClientBuilder(serviceClients = {})
+class SearchClientBuilder {
```

Comment:
Would it make more sense to use composition instead of inheritance for this builder helper class, we generally do not want public API inheriting from non-public API.

Reply:
yes, I cam move this to a composition approach. Wanted to avoid duplicating code between the 2  builders, hance this helper class.